config/v1/types.go

```diff
@@ -310,3 +310,91 @@ type DelegatedAuthorization struct {
+	// Defaults to the empty LabelSelector, which matches everything.
+	// A domain MUST be specified in order for namespaceSelector to apply
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,1,opt,name=namespaceSelector"`
+
+	// domainPatterns is an optional list of domains for which the desired HSTS annotations are required.
+	// If domainPatterns is specified and a route is created with a spec.host matching one of the domains,
+	// the route must specify the HSTS Policy components described in the matching RequiredHSTSPolicy.
+	// If domainPatterns is empty, the specified HSTS Policy components are IGNORED.
+	//
+	// The use of wildcards is allowed like this: *.foo.com matches everything under foo.com.
+	// foo.com only matches foo.com, so to cover foo.com and everything under it, you must specify *both*.
+	// +optional
+	DomainPatterns []string `json:"domainPatterns,omitempty" protobuf:"bytes,2,rep,name=domainsPattern"`
+
+	// maxAge is the delta time range in seconds during which hosts are regarded as HSTS hosts.
+	// If set to 0, it negates the effect, and hosts are removed as HSTS hosts.
+	// If set to 0 and includeSubdomains is specified, all subdomains of the host are also removed as HSTS hosts.
+	// maxAge is a time-to-live value, and if this policy is not refreshed on a client, the HSTS
+	// policy will eventually expire on that client.
+	// +required
+	MaxAge MaxAgePolicy `json:"maxage" protobuf:"bytes,3,opt,name=maxage"`
+
+	// preloadPolicy directs the client to include hosts in its host preload list so that
+	// it never needs to do an initial load to get the HSTS header (note that this is not defined
+	// in RFC 6797 and is therefore client implementation-dependent).
+	// +optional
+	PreloadPolicy PreloadPolicy `json:"preloadPolicy,omitempty" protobuf:"bytes,4,opt,name=preloadPolicy"`
+
+	// includeSubDomainsPolicy means the HSTS Policy should apply to any subdomains of the host's
+	// domain name.  Thus, for the host bar.foo.com, if includeSubDomainsPolicy was set to RequireIncludeSubDomains:
+	// - the host app.bar.foo.com would inherit the HSTS Policy of bar.foo.com
+	// - the host bar.foo.com would inherit the HSTS Policy of bar.foo.com
+	// - the host foo.com would NOT inherit the HSTS Policy of bar.foo.com
+	// - the host def.foo.com would NOT inherit the HSTS Policy of bar.foo.com
+	// +optional
+	IncludeSubDomainsPolicy IncludeSubDomainsPolicy `json:"includeSubDomainsPolicy,omitempty" protobuf:"bytes,5,opt,name=includeSubDomainsPolicy"`
+}
+
+// MaxAgePolicy contains a range for specifying a compliant HSTS max-age for the enclosing RequiredHSTSPolicy
+type MaxAgePolicy struct {
+	// The largest value of the RequiredHSTSPolicy max-age
+	// -1 means no opinion
+	// kubebuilder:validation:default=-1
+	LargestMaxAge int32 `json:"largestMaxAge,omitempty" protobuf:"varint,1,opt,name=largestMaxAge"`
```

Comment:
What does zero mean for this field?

Reply:
0 means the client should not use HSTS and should discard any HSTS policy that was previously set for the domain.